cql3/statements/modification_statement.cc

```diff
@@ -394,6 +403,9 @@ modification_statement::do_execute(service::storage_proxy& proxy, service::query
     }
 
     inc_cql_stats();
+    if (::is_system_keyspace(keyspace())) {
+        inc_cql_system_stats();
+    }
 
```

Comment:
It's bad form to slow down stuff just for stats. A single increment instruction now has to go through multiple string compares.

You can compute a pointer during prepare() and increment via the pointer.

Another question is whether a system read is a read of the system keyspace, or a read that the system executes for its internal purposes. It's even not clear what "system" is.

 - should writes to the traces tables be considered system writes? I think so
 - should driver queries of the system tables to get the schema be considered system reads? I think not

/cc @glommer who eats metrics for one meal a day

Reply:
Another way to address this is to add a pointer to the keyspace at prepare time, increment a keyspace metric, and have the metric fetch lambda sum over all system keyspaces. That's if we want by keyspace rather than by caller.

Reply:
@avikivity some of the questions you raised are inluded in the issue description https://github.com/scylladb/scylla/issues/4182#issuecomment-535011816

- we do need to count the driver queries to system tables as they are misleading our users in the cql_optimization dahsboard - full info in the issue

- we do need to split put the internal system read/writes as well